### ISSUE TITLE
mmm-facial-recognition-ocv3 Recognizes the User But Hotword not listening and activating the Assistant

### ISSUE THREAD

**Issue by aizazhussain341** [eouia/MMM-Hotword#103]
Connecting socket for: MMM-Facial-Recognition-OCV3
Starting module helper: MMM-Facial-Recognition-OCV3
Connecting socket for: MMM-Hotword
[HOTWORD] MMM-Hotword starts
Connecting socket for: MMM-AssistantMk2
Sockets connected & modules started ...
Launching application.
Create new calendar fetcher for url: https://calendar.google.com/calendar/ical/aizaz.hussain341%40gmail.com/public/basic.ics - Interval: 300000
Create new news fetcher for url: https://www.thenews.com.pk/rss/1/1 - Interval: 300000
[AMK2] Recipe is loaded: reboot.js
[AMK2] Recipe is loaded: screen_onoff.js
[AMK2] Recipe is loaded: hide_and_show_all_modules.js
[HOTWORD] begins.
[HOTWORD] Detector starts listening.
[AMK2] Temporal storage directory is clearing.
[MMM-Facial-Recognition-OCV3] Facerecognition started...
[MMM-Facial-Recognition-OCV3] Loading training data...
[HOTWORD] Detected: smartmirror
[HOTWORD] stops.
[MMM-Facial-Recognition-OCV3] Training data loaded!
[MMM-Facial-Recognition-OCV3] --------------------
[MMM-Facial-Recognition-OCV3] Webcam loaded...
[HOTWORD] Final Result: { detected: true, hotword: 'smartmirror', file: null }
[AMK2] assistant ready
[AMK2] Conversation starts.
[HOTWORD] begins.
[HOTWORD] Detector starts listening.
[AMK2] Assistant Text Response: 
[AMK2] Conversation Error: { Error: 13 INTERNAL: Received RST_STREAM with error code 0
    at Object.exports.createStatusError (/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/grpc/src/common.js:91:15)
    at ClientDuplexStream._emitStatusIfDone (/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/grpc/src/client.js:233:26)
    at ClientDuplexStream._receiveStatus (/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/grpc/src/client.js:211:8)
    at Object.onReceiveStatus (/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/grpc/src/client_interceptors.js:1306:15)
    at InterceptingListener._callNext (/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/grpc/src/client_interceptors.js:568:42)
    at InterceptingListener.onReceiveStatus (/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/grpc/src/client_interceptors.js:618:8)
    at /home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/grpc/src/client_interceptors.js:1123:18
  code: 13,
  metadata: Metadata { _internal_repr: {}, flags: 0 },
  details: 'Received RST_STREAM with error code 0' }
[MMM-Facial-Recognition-OCV3] User Aizaz with confidence 63.52510035679746 logged in.
[HOTWORD] stops.
[HOTWORD] Final Result: { detected: false }
[HOTWORD] begins.
[HOTWORD] Detector starts listening.
[AMK2] assistant ready
[AMK2] Conversation starts.
[AMK2] Assistant Text Response:

**Comment by eouia** [comment]
Hmmm.. Maybe you are using webcam as mic. So, during your Facial Recognition working, mic is operated together, and that seems making this issue.

**Comment by aizazhussain341** [comment]
I am using Google Voive Kit VoiceHat Microphone for hotword detection and Logitec Webcam for Facial Recognition.

**Comment by eouia** [comment]
Hmm. can you show me your configuration?

**Comment by aizazhussain341** [comment]
/* Magic Mirror Config Sample
*
* By Michael Teeuw http://michaelteeuw.nl
* MIT Licensed.
*
* For more information how you can configurate this file
* See https://github.com/MichMich/MagicMirror#configuration
*
*/

var config = {
	address: "localhost", // Address to listen on, can be:
	// - "localhost", "127.0.0.1", "::1" to listen on loopback interface
	// - another specific IPv4/6 to listen on a specific interface
	// - "", "0.0.0.0", "::" to listen on any interface
	// Default, when address config is left out, is "localhost"
	port: 8080,
	ipWhitelist: ["127.0.0.1", "::ffff:127.0.0.1", "::1"], // Set [] to allow all IP addresses
	// or add a specific IPv4 of 192.168.1.5 :
	// ["127.0.0.1", "::ffff:127.0.0.1", "::1", "::ffff:192.168.1.5"],
	// or IPv4 range of 192.168.3.0 --> 192.168.3.15 use CIDR format :
	// ["127.0.0.1", "::ffff:127.0.0.1", "::1", "::ffff:192.168.3.0/28"],

	language: "en",
	timeFormat: 24,
	units: "metric",

	modules: [
		{
			module: "alert",
		},
		{
			module: "updatenotification",
			position: "top_bar"
		},
		{
			module: "clock",
			position: "top_left",
			classes:'Mudasir',
		},
		{
			module: "calendar",
			header: "My Events",
			position: "top_center",
			classes: 'Aizaz',
			config: {
				colored: false,
				coloredSymbolOnly: false,
				calendars: [
					{
						url: "https://calendar.google.com/calendar/ical/aizaz.hussain341%40gmail.com/public/basic.ics",
						symbol: "calendar"
					}
				]
			}
		},
		{
			module: "compliments",
			position: "lower_third"
		},
		{
			module: "currentweather",
			classes:'Aizaz',
			position: "top_right",
			header: "Today`s Weather",
			config: {
				location: "Bahawalpur",
				locationID: "1183880", //ID from http://bulk.openweathermap.org/sample/; unzip the gz file and find your city
				appid: "62c6a27b8a3e8a8c4562a974cf457aac"
			}
		},
		{
			module: "weatherforecast",
			position: "top_right",
			classes: 'Mudasir',
			header: "Weather Forecast",
			config: {
				location: "Bahawalpur",
				locationID: "1183880", //ID from http://www.openweathermap.org/help/city_list.txt
				appid: "62c6a27b8a3e8a8c4562a974cf457aac"
			}
		},
		{
			module: "newsfeed",
			position: "bottom_bar",
			classes: 'Aizaz',
			config: {
				feeds: [
					{
						title: "The News",
						url: "https://www.thenews.com.pk/rss/1/1"
					}
				],
				showSourceTitle: true,
				showPublishDate: true
			}
		},

		// {
		// 	module: 'MyQrScanner',
		// 	position:'bottom_left',
		// 	classes: 'Aizaz',
		// 	config:{
		// 		text: "Aizaz",
		// 		width: 100,
		// 		height: 100,
		// 	}
		// },
		// {
		// 	module: "MMM-GAction",
		// 	config: {

		// 	}
		// },



		// {
		// 	module: "helloworld",	
		// 	position: "bottom_bar",	// This can be any of the regions.
		// 	config: {
		// 		// See 'Configuration options' for more information.
		// 		text: "Hello world!"
		// 	}
		// },

		// {
		// 	module: 'MMM-RandomQuranAyah',
		// 	position: 'top_bar',	// This can be any of the regions. Best result is in the top_bar/bottom_bar as ayah (verse) can take multiple lines.
		// 	config: {
		// 		apiVersion: '1.0', // please, leave unchanged. reserved for future use.
		// 		showArabic: true,
		// 		showTranslation: true,
		// 		surahArabicName:false,
		// 		translationLang:'id.indonesian',
		// 		updateInterval: 3600 * 1000, // milliseconds
		// 	}
		// },
		{
			module: 'MMM-Facial-Recognition-OCV3',
			config: {
				// Threshold for the confidence of a recognized face before it's considered a
				// positive match.  Confidence values below this threshold will be considered
				// a positive match because the lower the confidence value, or distance, the
				// more confident the algorithm is that the face was correctly detected.
				threshold: 80,
				// force the use of a usb webcam on raspberry pi (on other platforms this is always true automatically)
				useUSBCam: true,
				// Path to your training xml
				trainingFile: '/home/pi/MagicMirror/modules/MMM-Facial-Recognition-OCV3/training.xml',
				// recognition intervall in seconds (smaller number = faster but CPU intens!)
				interval: 300,
				// Logout delay after last recognition so that a user does not get instantly logged out if he turns away from the mirror for a few seconds
				logoutDelay: 5,
				// Array with usernames (copy and paste from training script)
				users: ["Aizaz"],
				//Module set used for strangers and if no user is detected
				defaultClass: "default",
				//Set of modules which should be shown for every user
				everyoneClass: "everyone",
				// Boolean to toggle welcomeMessage
				welcomeMessage: true
			}
		},

		{
			module: "MMM-Hotword",
			position: "bottom_left",
			classes: 'Aizaz',
			config: {
				chimeOnFinish: null,
				autostart: true, // if 'false', this module will wait for 'HOTWORD_RESUME' notification to start hotwords detection at the beginning.
				autorestart: true,

				models: [
					{
						hotwords: "smartmirror",
						file: "smart_mirror.umdl",
						sensitivity: "0.5",
					},
				],
				commands: {
					"smartmirror": {
						notificationExec: {
							notification: "ASSISTANT_ACTIVATE",
							payload: (detected, afterRecord) => {
								return { profile: "default" };
							}
						},
						restart: true,
						afterRecordLimit: 0
					}
				}
			}
		},
		
		// {
		// 	module: 'newQrCodeScanner',
		// 	position:'top_right',
		// 	config:{
		// 		text: "Aizaz",
		// 		width: 100,
		// 		height: 100,
		// 	}
		// },
		// {
		// 	module: 'MMM-PrayerTime',
		// 	position: 'center',	// This can be any of the regions. Best result is in the top_left/top_right.
		// 	config: {
		// 	apiVersion: '1.0', // please, leave unchanged. reserved for future use.
		// 	lat: 29.418068, // latitude of your position (city)
		// 	lon: 71.670685, // longitude of your position (city)
		// 	timezone: "Asia/Karachi",
		// 	timeFormat: 24,
		// 	method: 3,
		// 	playAdzan: ['fajr', 'dhuhr', 'asr', 'maghrib', 'isha'],
		// 	notDisplayed: ['midnight', 'sunset'],
		// 	useUpdateInterval: true,
		// 	updateInterval: 86400 * 1000, // How often do you want to fetch new praying time? (milliseconds)
		// 	animationSpeed: 2.5 * 1000, // Speed of the update animation. (milliseconds)
		// 	language: 'en',
		// 	showAdzanAlert: true,
		// 	showTomorrow: false,
		// 	vertical: true, // set false for horizontal view
		// 	alertTimer: 20000,
		// }
		// },
		{
			module: "MMM-AssistantMk2",
			position: "bottom_right",
			// useGactionCLI: true,
			// projectId: "mymagicmirror-f493d",
			
			classes: 'Aizaz',
			config: {
				
				deviceLocation: {
					coordinates: { // set the latitude and longitude of the device to get localized information like weather or time. (ref. mygeoposition.com)
						latitude: 29.418068, // -90.0 - +90.0
						longitude: 71.670685, // -180.0 - +180.0
						
					},
				},
				notifications: {
					ASSISTANT_ACTIVATED: "HOTWORD_PAUSE",
					ASSISTANT_DEACTIVATED: "HOTWORD_RESUME",
				},
				recipes: ["reboot.js", "screen_onoff.js","hide_and_show_all_modules.js",],
				// action: {
				// 	"com.example.commands.REBOOT" : {
				// 	  command: "REBOOT"
				// 	},
				// 	"com.example.commands.PAGE" : {
				// 	  command: "PAGE"
				// 	},
				//   },
				  
				//   command: {
				// 	"PAGE": {
				// 	  notification:(params)=>{
				// 		if (params.number) {
				// 		  return "PAGE_SELECT"
				// 		} else if (params.incordec == "INC") {
				// 		  return "PAGE_INCREMENT"
				// 		} else {
				// 		  return "PAGE_DECREMENT"
				// 		}
				// 	  },
				// 	  payload:()=>{
				// 		if (params.number) {
				// 		  return params.number
				// 		} else {
				// 		  return null
				// 		}
				// 	  }
				// 	},
				//   },
				pauseOnYoutube: false,
				// useWelcomeMessage: "brief today",
				profiles: {
					"default": {
						lang: "en-US"
					}
				},
				
			},
			

			
		},
	]

};

/************* DO NOT EDIT THE LINE BELOW *************/
if (typeof module !== "undefined") {
	module.exports = config;
}

**Comment by aizazhussain341** [comment]
it works on system but not working on Raspberry pi.

**Comment by eouia** [comment]
```js
commands: {
				"smartmirror": {
					notificationExec: {
						notification: "ASSISTANT_ACTIVATE",
						payload: (detected, afterRecord) => {
							return { profile: "default" };
						}
					},
					restart: true,
					afterRecordLimit: 0
				}
			}
```
set `restart: false`,

**Comment by aizazhussain341** [comment]
Still doesn`t work
Here`s the Log
Sockets connected & modules started ...
Launching application.
Create new calendar fetcher for url: https://calendar.google.com/calendar/ical/aizaz.hussain341%40gmail.com/public/basic.ics - Interval: 300000
Create new news fetcher for url: https://www.thenews.com.pk/rss/1/1 - Interval: 300000
[AMK2] Recipe is loaded: reboot.js
[AMK2] Recipe is loaded: screen_onoff.js
[AMK2] Recipe is loaded: hide_and_show_all_modules.js
[AMK2] Temporal storage directory is clearing.
[HOTWORD] begins.
[HOTWORD] Detector starts listening.
[MMM-Facial-Recognition-OCV3] Facerecognition started...
[MMM-Facial-Recognition-OCV3] Loading training data...
[MMM-Facial-Recognition-OCV3] Training data loaded!
[MMM-Facial-Recognition-OCV3] --------------------
[MMM-Facial-Recognition-OCV3] Webcam loaded...
[MMM-Facial-Recognition-OCV3] User Aizaz with confidence 62.51291769945522 logged in.
[MMM-Facial-Recognition-OCV3] User undefined with confidence null logged in.
[MMM-Facial-Recognition-OCV3] User undefined logged out.
[MMM-Facial-Recognition-OCV3] User undefined with confidence null logged in.
[MMM-Facial-Recognition-OCV3] User undefined logged out.
[MMM-Facial-Recognition-OCV3] User Aizaz with confidence 58.565417458746886 logged in.
[MMM-Facial-Recognition-OCV3] User Aizaz logged out.
[MMM-Facial-Recognition-OCV3] User undefined with confidence null logged in.

**Comment by aizazhussain341** [comment]
it recognizes me but not listening the Hotword

**Comment by eouia** [comment]
To make things simple, just test without Facial-Recognition module?

**Comment by aizazhussain341** [comment]
When i run without MMM-Facial-recognition-ocv3 it detects hotword and activates the Assistant

**Comment by aizazhussain341** [comment]
I want to run along with MMM-Facial-recognition-ocv3.

**Comment by aizazhussain341** [comment]
Could you please help?

**Comment by eouia** [comment]
I don't know exactly what collision would have happened, but it looks, your environment get audio stream from webcam not your dedicated mic when FR module is activated. Probably it might be ALSA setup issue, However, I'm not that kind of Expert so can't help you.

**Comment by aizazhussain341** [comment]
Thanks for the help It worked for me when i disabled the mic of webcam.
Thanks
